### ISSUE TITLE
[interop] C++ function returning `NonCopyable` type returns `Never` in Swift and is callable

### ISSUE THREAD

**Issue by hyp** [apple/swift#64401]
This code:

```c++
struct NonCopyable {
    NonCopyable(const NonCopyable &) = delete;
    
    int x;
};

NonCopyable returnsNoncopyable();
```

Produces callable `returnsNoncopyable` function in Swift module interface that returns `Never`:

```swift
func returnsNoncopyable() -> Never
```

This function is callable, and doesn't result in an error unless the return value is attempted to be used.

**Comment by Rajveer100** [comment]
@hyp Could you give a hint on how one would go about this?

**Comment by Rajveer100** [comment]
@AnthonyLatsis Are there any resources I can go through to solve this?

**Comment by AnthonyLatsis** [comment]
My best guess would be that this is an issue with `ClangImporter`. My interoperability kung fu is pretty lame though, so take this with a grain of salt. That aside, C++ interop is an experimental feature. If I were to poke at this, I would skim through the [manifesto](https://github.com/apple/swift/blob/main/docs/CppInteroperability/CppInteroperabilityManifesto.md) first.

**Comment by Rajveer100** [comment]
Well, ok!
@egorzhdan What's your take on this! Please let me know!

**Comment by hyp** [comment]
Sorry for delay, I already have a PR that addresses on this issue: https://github.com/apple/swift/pull/65105 . Not yet merged to 5.9 though

**Comment by hyp** [comment]
Will work around it in a different way
